qualification/Jenkinsfile

```diff
@@ -73,20 +73,33 @@ pipeline {
          script {
            sh 'pip install -r qualification/requirements.txt'
            sh 'pip install --no-deps ".[qualification]" && pip check'
-           try {
-             sh 'spead2_net_raw pytest -v -c qualification/pytest-jenkins.ini qualification --image-override katgpucbf:harbor.sdp.kat.ac.za/cbf/katgpucbf:latest'
-           }
-           finally {
-             COMMIT_ID = sh(script: 'qualification/report/generate_pdf.py report.json report.pdf -c', returnStdout: true)
-             sh(script: 'qualification/report/generate_pdf.py report.json procedure.pdf --generate-procedure-doc')
-             publishHTML(target: [keepAll: true, reportName: 'Qualification Test Report', reportDir: '', reportFiles: 'report.pdf, report.json', reportTitles: 'report.pdf, report.json'])
-             publishHTML(target: [keepAll: true, reportName: 'Qualification Test Procedure', reportDir: '', reportFiles: 'procedure.pdf', reportTitles: 'procedure.pdf'])
-             currentBuild.displayName = "#${BUILD_NUMBER} (${COMMIT_ID})"
-           }
+           sh 'spead2_net_raw pytest -v -c qualification/pytest-jenkins.ini qualification --suppress-tests-failed-exit-code --image-override katgpucbf:harbor.sdp.kat.ac.za/cbf/katgpucbf:latest --junitxml=result.xml'
          }
        }
-    }
-  }
+     }
+
+     stage('Publish results') {
+       steps {
+         script {
+           // Compress and publish json file so that at least we have that.
+           sh 'xz --keep --force -T 0 report.json'
+           publishHTML(target: [keepAll: true, reportName: 'Qualification Test Intermediate JSON', reportDir: '', reportFiles: 'report.json.xz', reportTitles: 'report.json.xz'])
+
+           // Generate and publish test report
+           COMMIT_ID = sh(script: 'qualification/report/generate_pdf.py report.json report.pdf -c', returnStdout: true)
+           currentBuild.displayName = "#${BUILD_NUMBER} (${COMMIT_ID})"
+           publishHTML(target: [keepAll: true, reportName: 'Qualification Test Report', reportDir: '', reportFiles: 'report.pdf', reportTitles: 'report.json'])
+
+           // If that worked, we can probably generate and publish a procedure as well
+           sh(script: 'qualification/report/generate_pdf.py report.json procedure.pdf --generate-procedure-doc')
+           publishHTML(target: [keepAll: true, reportName: 'Qualification Test Procedure', reportDir: '', reportFiles: 'procedure.pdf', reportTitles: 'procedure.pdf'])
+
+           // Since everything is generated, we don't need the original anymore
+           sh 'rm -f report.json'
```

Comment:
We were getting the error of `Already exists` previously on this step, which suggested to me that the cleanup wasn't actually happening.

Reply:
I'm guessing that was because you had stuff in `post` that was failing, causing the rest of the `post` block to fail.

From the documentation it looks like you can have an additional `cleanup` block that follows `post`.

Reply:
Ok, I just pushed a change removing this. I'm going to wait for the qual test to run, and then try and examine what's actually going on inside the Jenkins container to see whether it did clean up.